src/sst/core/configGraph.cc

```diff
@@ -421,6 +481,43 @@ ConfigComponent* ConfigComponent::findSubComponentByName(const std::string& name
     return nullptr;
 }
 
+ConfigStatistic* ConfigComponent::insertStatistic(StatisticId_t sid)
+{
+  ConfigComponent* parent = getParent();
+  if (parent){
+    return parent->insertStatistic(sid);
+  } else {
+    return &statistics[sid];
+  }
+}
+
+ConfigStatistic* ConfigComponent::findStatistic(const std::string& name) const
+{
+  auto iter = enabledStatNames.find(name);
+  if (iter != enabledStatNames.end()){
+    StatisticId_t id = iter->second;
+    return findStatistic(id);
+  } else {
+    return nullptr;
+  }
+}
+
+ConfigStatistic* ConfigComponent::findStatistic(StatisticId_t sid) const
+{
+    auto* parent = getParent();
+    if (parent){
+      return parent->findStatistic(sid);
+    } else {
+      auto iter = statistics.find(sid);
+      if (iter == statistics.end()){
+        return nullptr;
+      } else {
+        //I hate that I have to do this
+        return const_cast<ConfigStatistic*>(&iter->second);
```

Comment:
The method is const, so we cannot use it to return a pointer that allows the modification of a member variable of the class. &iter->second refers to the statistic member of the class, which is here const because the method is const.
That's why we need to const_cast it.

Reply:
In that case, it seems like the function shouldn't be marked const.  I assume that this function is called some place on a ConfigComponent that is marked const??